src/analysis/normed_space/weak_dual_of_normed_space.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2021 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä, Heather Macbeth
+-/
+import topology.algebra.weak_dual_topology
+import analysis.normed_space.dual
+import analysis.normed_space.operator_norm
+
+/-!
+# Weak dual of normed space
+
+This file proves properties of the weak-* topology on the dual of a normed space.
+
+It is shown that the canconical mapping `dual 𝕜 E → weak_dual 𝕜 E` is continuous, and as a
+consequence the weak-* topology is coarser than the topology induced by the dual norm (operator
+norm).
+
+## Main definitions
+
+The main definitions concern the canconical mapping `dual 𝕜 E → weak_dual 𝕜 E`.
+
+* `normed_space.dual.to_weak_dual` is a linear equivalence from `dual 𝕜 E`to `weak_dual 𝕜 E`.
+* `normed_space.dual.continuous_linear_map_to_weak_dual` is a continuous linear mapping from
+  `dual 𝕜 E` to `weak_dual 𝕜 E`.
+
+## Main results
+
+The main results concern the comparison of the operator norm topology on `dual 𝕜 E` and the weak-*
+topology on (it type synonym) `weak_dual 𝕜 E`.
+
+* `dual_norm_topology_le_weak_dual_topology` is the statement that the weak-* topology on the dual
+  of a normed space is coarser (not necessarily strictly) than the operator norm topology.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+Weak-* topology is defined generally in the file <topology.algebra.weak_dual_topology>.
+
+When `E` is a normed space, the duals `dual 𝕜 E` and `weak_dual 𝕜 E` are type synonyms with
+different topology instances.
+
+## References
+
+* https://en.wikipedia.org/wiki/Weak_topology#Weak-*_topology
+
+## Tags
+
+weak-star, weak dual
+
+-/
+
+noncomputable theory
+open filter
+open_locale topological_space
+
+section weak_star_topology_for_duals_of_normed_spaces
+/-!
+### Weak star topology on duals of normed spaces
+In this section, we prove properties about the weak-* topology on duals of normed spaces.
+We prove in particular that the canonical mapping `dual 𝕜 E → weak_dual 𝕜 E` is continuous,
+i.e., that the weak-* topology is coarser (not necessarily strictly) than the topology given
+by the dual-norm (i.e. the operator-norm).
+-/
+
+open normed_space
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+/-- For normed spaces `E`, there is a canonical map `dual 𝕜 E → weak_dual 𝕜 E` (the "identity"
+    mapping). It is a linear equivalence. -/
+def normed_space.dual.to_weak_dual : dual 𝕜 E ≃ₗ[𝕜] weak_dual 𝕜 E :=
+linear_equiv.refl 𝕜 (E →L[𝕜] 𝕜)
+
+/-- For normed spaces `E`, there is a canonical map `weak_dual 𝕜 E → dual 𝕜 E` (the "identity"
+    mapping). It is a linear equivalence. Here it is implemented as the inverse of the linear
+    equivalence `normed_space.dual.to_weak_dual` in the other direction. -/
+def weak_dual.to_original_dual : weak_dual 𝕜 E ≃ₗ[𝕜] dual 𝕜 E :=
```

Comment:
Wondering about the word `original`, because the weak dual is in some sense a more elementary concept, lower in the hierarchy ... what do you think of `to_normed_dual` or `to_strong_dual`?  (I am never quite sure what "strong topology" is supposed to mean.)

Reply:
I fully agree, despite having introduced the term here...

I had actually initially called this "ordinary" rather than "original", but decided to check Rudin's "Functional Analysis" --- in the hopes of finding a better and established term. Strictly speaking, Rudin doesn't (seem to) have terminology for the dual with its usual operator norm topology, but in contrasting the weak (not weak-*) topology on a topological vector space to its usual topology, he uses the term "original". (He has a footnote saying that "strong" is common synonym for his "original", but makes an argument to avoid it in that context.) I picked the term in my context as it sounded rather neutral and safe (not good reasons)...

Perhaps some of the main criteria for good terminology would be that it is descriptive, established, and unambiguous. I think "original" is not particularly convincing about any of these (although not totally terrible w.r.t. any of them either?).

Your "normed dual" is certainly descriptive and unambiguous, so it seems very good to me! I am not totally sure, but once one generalizes the theory to reasonable topological vector spaces, "strong topology" on the dual might be rather standard for something that specializes to "dual norm topology" for the duals of normed spaces. For now I would not pick it because (1) I'm not 100% sure about what I just said, (2) "strong" certainly suffers from at least a bit of ambiguity in some fields of math, and (3) it should be easy to change later when the theory in mathlib is generalized to topological vector spaces (which will almost surely happen in a finite time, right?).

So I would go with "normed_dual" for now, and will leave it to those who generalize the theory to change it appropriately.

Thanks for the improvement!

Reply:
Oh!  Sorry to have made you switch, I think anything backed by the authority of Rudin is also fine :)  Do what you think best (or what is easiest, given you already switched to `normed_dual` ...)

Reply:
No, I think both of your suggestions were strictly better than what I used. I picked the safer looking one of them, but look forward to the day when it will look like a special case of something more general and has to be renamed as your other suggestion. I was at no point backed by the authority of Rudin, since I extrapolated Rudin's terminology from a different context --- Rudin's actual choice seems obviously more thoughtful than the extrapolation.